src/core/document.js

```diff
@@ -125,11 +135,13 @@ class Page {
     // For robustness: The spec states that a \Resources entry has to be
     // present, but can be empty. Some documents still omit it; in this case
     // we return an empty dictionary.
-    return shadow(
-      this,
-      "resources",
-      this._getInheritableProperty("Resources") || Dict.empty
-    );
+
+    const resources = this._getInheritableProperty("Resources") || Dict.empty;
+    const acroForm = this.pdfManager.pdfDocument.catalog.acroForm;
+    if (acroForm && acroForm.get("DR")) {
+      resources.defaultResources = acroForm.get("DR");
```

Comment:
Hm, I wouldn't expect the normal resources dictionary to be altered with a non-standard attribute. Shouldn't we merge the dictionaries instead? Perhaps the just merged https://github.com/mozilla/pdf.js/pull/12361/files can serve as inspiration.